From: Shriram Ravindranathan <s20n@ters.dev>
Date: Sun, 3 May 2026 19:09:27 +0530
Subject: Fix reproducible builds

chibi-scheme-doc:
Suppressed memory addresses in documentation and timing metrics
in test outputs when SOURCE_DATE_EPOCH is set. This ensures the
chibi-scheme-doc package builds reproducibly.

chibi-scheme-common:
Sorted find output in Makefile for reproducible builds.
The generated .chibi.meta, .scheme.meta, and .srfi.meta files had
non-deterministic ordering on rebuilderd causing FTBR. This patch
fixes those issues by piping find output through LC_ALL=C sort.
See: https://reproduce.debian.net/all/api/v1/builds/202656/artifacts/414510/diffoscope

Forwarded: not-needed
Author: Shriram Ravindranathan <s20n@ters.dev>
Last-Update: 2026-05-17
---
 Makefile           | 3 ++-
 lib/chibi/csv.scm  | 7 ++++---
 lib/chibi/doc.scm  | 4 +---
 lib/chibi/test.scm | 4 +++-
 4 files changed, 10 insertions(+), 8 deletions(-)

diff --git a/Makefile b/Makefile
index 2967f22..4ffad3a 100644
--- a/Makefile
+++ b/Makefile
@@ -154,7 +154,7 @@ clibs.c: $(CHIBI_GENSTATIC_DEPENDENCIES) $(COMPILED_LIBS:%$(SO)=%.c)
 	if [ -d .git ]; then \
 		$(GIT) ls-files lib | $(GREP) .sld | $(CHIBI_GENSTATIC) > $@; \
 	else \
-		$(FIND) lib -name \*.sld | $(CHIBI_GENSTATIC) > $@; \
+		$(FIND) lib -name \*.sld | LC_ALL=C sort | $(CHIBI_GENSTATIC) > $@; \
 	fi
 
 chibi-scheme.pc: chibi-scheme.pc.in
@@ -195,6 +195,7 @@ doc: doc/chibi.html doc-libs
 
 lib/.%.meta: lib/%/ tools/generate-install-meta.scm $(CHIBI_DEPENDENCIES)
 	-$(FIND) $< -name \*.sld | \
+	 LC_ALL=C sort | \
 	 $(CHIBI) tools/generate-install-meta.scm $(CHIBI_VERSION) > $@
 
 ########################################################################
diff --git a/lib/chibi/csv.scm b/lib/chibi/csv.scm
index d5c572d..f2967e5 100644
--- a/lib/chibi/csv.scm
+++ b/lib/chibi/csv.scm
@@ -37,9 +37,10 @@
 ;;> Example Gecos grammar:
 ;;>
 ;;> \example{
-;;> (csv-grammar
-;;>   '((separator-chars #\\:)
-;;>     (quote-char . #f)))
+;;> (csv-grammar?
+;;>   (csv-grammar
+;;>     '((separator-chars #\\:)
+;;>       (quote-char . #f))))
 ;;> }
 (define (csv-grammar spec)
   (let ((grammar (make-csv-grammar '(#\,) #\" #t #f 'lax '() #f #f)))
diff --git a/lib/chibi/doc.scm b/lib/chibi/doc.scm
index c084704..db967d8 100644
--- a/lib/chibi/doc.scm
+++ b/lib/chibi/doc.scm
@@ -343,9 +343,7 @@
 
 (define (expand-author sxml env)
   `(div (@ (id . "notes"))
-        ,@(map (lambda (x) (expand-docs x env)) (cdr sxml))
-        (br)
-        ,(seconds->string (current-seconds))))
+        ,@(map (lambda (x) (expand-docs x env)) (cdr sxml))))
 
 (define (expand-code sxml env)
   (let* ((hl (if (and (pair? (cdr sxml)) (eq? 'language: (cadr sxml)))
diff --git a/lib/chibi/test.scm b/lib/chibi/test.scm
index e26516e..0a57997 100644
--- a/lib/chibi/test.scm
+++ b/lib/chibi/test.scm
@@ -802,7 +802,9 @@
         (display
          (string-append
           (plural " test" pass) " passed in "
-          (number->string duration) " seconds"
+          (if (get-environment-variable "SOURCE_DATE_EPOCH")
+              ""
+              (string-append " in " (number->string duration) " seconds"))
           (cond
            ((zero? skip) "")
            (else (string-append " (" (number->string skip)
